### PR TITLE
Fix test: handler losing connectivity

### PR DESCRIPTION
**Pull request by xpivarc**
**What this PR does / why we need it**:
Fixing leak of the test that affect [[test_id:2666]
](https://github.com/kubevirt/kubevirt/pull/6001). 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: tests/virt_control_plane_test.go [hunk 1]
```diff
@@ -256,6 +256,10 @@ var _ = Describe("[Serial][ref_id:2717][sig-compute]KubeVirt control plane resil
 			}
 
 			It("should fail health checks when connectivity is lost, and recover when connectivity is regained", func() {
+				daemonSet, err := virtCli.AppsV1().DaemonSets(flags.KubeVirtInstallNamespace).Get(context.Background(), componentName, metav1.GetOptions{})
```
Comment (fgimenez) [inline tests/virt_control_plane_test.go:259]: This call is repeated in `readyFunc`, maybe we could refactor?

File: tests/virt_control_plane_test.go [hunk 2]
```diff
@@ -269,7 +273,7 @@ var _ = Describe("[Serial][ref_id:2717][sig-compute]KubeVirt control plane resil
 				blackHolePodFunc("del")
 
 				By("ensuring we now have a ready virt-handler daemonset")
-				Eventually(readyFunc, 30*time.Second, time.Second).Should(BeNumerically(">", 0))
+				Eventually(readyFunc, 30*time.Second, time.Second).Should(BeNumerically("==", desiredDeamonsSetCount))
```
Comment (fgimenez) [inline tests/virt_control_plane_test.go:276]: Will 30s be enough to have `desiredDaemonsSetCount` in ready state?

**Review by fgimenez — COMMENTED** [review]
Great catch, thanks a lot!

File: tests/virt_control_plane_test.go (hunk 1 above)
Comment (xpivarc) [reply tests/virt_control_plane_test.go:259]: I am not sure as I would then need to change the `Eventually`. I think 1 more call is not a problem.

File: tests/virt_control_plane_test.go (hunk 2 above)
Comment (xpivarc) [reply tests/virt_control_plane_test.go:276]: I don't have data to back it up but I feel like it should be enough

File: tests/virt_control_plane_test.go (hunk 1 above)
Comment (fgimenez) [reply tests/virt_control_plane_test.go:259]: No need to change the `Eventually`, you could create a separate function that returns the daemonset (checking also for the potential error), and call it from `readyFunc` and from the body of the `It`. Just a nit in any case, as you see fit.

Comment (xpivarc) [reply tests/virt_control_plane_test.go]: Done

**Review by fgimenez — APPROVED** [review]
/lgtm
/approve

Thanks!

**Pull request merged**